src/lein_monolith/dependency.clj

```diff
@@ -93,14 +93,28 @@
 
 ;; ## Dependency Graphs
 
+(defn- collect-dependencies
+  "Merges the project's top-level dependencies with all dependencies listed in
+  the project's profiles to ensure the project has the proper dependency closure
+  for compilation ordering."
+  [project]
+  (->>
+    project
+    (:profiles)
+    (vals)
+    (cons project)
+    (mapcat :dependencies)
+    (set)))
```

Comment:
The `set` here doesn't really do much given that you're immediately turning this into a further list of dependency name symbols and calling `set` on them again below. You could also pull that `(map (comp condense-name first) ...)` transformation into this function instead if you wanted.